src/kubeseal_auto/host.py

```
@@ -1,67 +1,235 @@
+                if member.name == "kubeseal" and member.isfile():
+                    # Create a modified member with the versioned name
+                    member.name = target_name
+                    tar.extract(member, path=self.bin_location, filter="data")
+                    return
+        else:
+            # Fallback for Python < 3.12: manual safe extraction
+            for member in tar.getmembers():
+                if member.name == "kubeseal" and member.isfile():
+                    # Security checks for path traversal
+                    if member.name.startswith("/") or ".." in member.name:
+                        raise ValueError(f"Unsafe path in tar archive: {member.name}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)